File: frontend/src/api/knowledge.ts
import api from './client'

export interface Insight {
  doc_id: string
  title: string
  content: string
  hash: string
  expandable?: boolean
}

export async function getInsights(): Promise<Insight[]> {
  const { data } = await api.get<Insight[]>('/knowledge/insights')
  return data
}

export async function generateInsights(): Promise<Insight[]> {
  const { data } = await api.post<Insight[]>('/knowledge/insights/generate')
  return data
}

File: backend/app/presentation/api/knowledge.py
from fastapi import APIRouter

from app.application.services.knowledge_service import KnowledgeService
from app.infrastructure.persistence.session import SessionLocal

router = APIRouter(prefix="/api/knowledge", tags=["knowledge"])


def _service() -> KnowledgeService:
    return KnowledgeService(SessionLocal)


@router.get("/insights")
async def get_insights():
    return _service().get_insights()


@router.post("/insights/generate")
async def generate_insights():
    return _service().generate_insights()
